### PR TITLE
moved the material on string constructors out of the string manipulat…

### PR DESCRIPTION
**Pull request by linda-seiter**
…ion lesson into a new separate lesson

### CODE REVIEW

**Comment by developerjaym** [comment]
This might be a good place to explain why '==' isn't great when comparing Strings.

**Review by developerjaym — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,32 +1,154 @@
-# String Constructors
+# String Manipulation
```
Comment (kcleland0818) [inline README.md]: I think this still needs to be called "String Constructors"

File: README.md [hunk 2]
```diff
@@ -1,32 +1,154 @@
...
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+```java
+String cat = "Tom";
+```
+
+This code uses a **string literal** "Tom" to assign a value to a `String` data type.
+
+But what happens when we assigned multiple variables to the same string literal,
```
Comment (kcleland0818) [inline README.md]: I think "assigned" should be present-tense, "assign"

File: README.md [hunk 3]
```diff
@@ -1,32 +1,154 @@
...
+a string literal that already exists on the heap.
+
+## String Constructor
+
+Now what if we want to have two different `String` objects with the same sequence of characters?
+Java's `String` class actually comes with several constructors, so we can use the `new`
+operator followed by a constructor call such as `String()`
+to help instantiate different `String` objects with the same character sequence!
+
+Recall that a constructor is a special
```
Comment (kcleland0818) [inline README.md]: I'm not sure constructors have been introduced at this point? So maybe remove the word "Recall"

File: README.md [hunk 4]
```diff
@@ -1,32 +1,154 @@
...
+
+To better explain some of these constructors, let's look at each one:
+
+1. `String cat1 = new String();` - The first constructor does not take any parameters - this will assign
+   the  variable `cat1` to a new `String` object that contains an empty character sequence; which we can show with two
+   double quotes with nothing inside them: `""`.
+2. `String cat2 = new String(name);` - The second constructor takes in a `String` as a parameter. This will
+   assign the variable `cat2` to a new `String` object that contains the same sequence of characters as the
+   `String` object referenced by `name`, i.e. "Tom-Cat".
+   - Notice that `name` and `cat` reference different objects in memory,
```
Comment (kcleland0818) [inline README.md]: Should it be `cat2`?

File: README.md (hunk 2 above)
Comment (linda-seiter) [reply README.md]: fixed

File: README.md (hunk 1 above)
Comment (linda-seiter) [reply README.md]: yes, thanks

File: README.md (hunk 3 above)
Comment (linda-seiter) [reply README.md]: I briefly describe constructors as a special method for creating objects in the introductory object/class lesson,
but yes probably best to remove "Recall".

File: README.md (hunk 4 above)
Comment (linda-seiter) [reply README.md]: yes, thanks

**Comment by linda-seiter** [comment]
> This might be a good place to explain why '==' isn't great when comparing Strings.

Yes good idea.  I added a section at the end.

File: README.md [hunk 5]
```diff
@@ -76,7 +76,7 @@ Java's `String` class actually comes with several constructors, so we can use th
 operator followed by a constructor call such as `String()`
 to help instantiate different `String` objects with the same character sequence!
 
-Recall that a constructor is a special
+A constructor is a special
 method used to initialize the fields of an object. A class can have several
 constructors as long as the number or type of the parameters differ.
```
Comment (kcleland0818) [inline README.md]: Typo (maybe from me?): "A class can have several constructors as long as the number or type of the parameters differ."

Remove the extra "the" in "or type of the parameters differ"

**Review by kcleland0818 — APPROVED** [review]
Great explanation on strings and `==`!

**Review by developerjaym — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (linda-seiter) [reply README.md]: yes, fixed

**Review by AlveeM — APPROVED** [review]

**Pull request merged**